ContextApp/src/main/java/com/github/migi_1/ContextApp/HeartsUpdateFunctions.java

```diff
@@ -0,0 +1,132 @@
+package com.github.migi_1.ContextApp;
+
+import android.widget.ImageView;
+import com.github.migi_1.ContextMessages.PlatformPosition;
+
+/**
+ * Contains functions which changes the hearts that represent the 
+ * player's health.
+ */
+public class HeartsUpdateFunctions {
+
+    private MainActivity act;
+    private PlatformPosition position;
+    
+    /**
+     * This creates an instance of the HeartsUpdateFunctions.
+     * @param act
+     * 			The instance of the application that calls this function
+     * @param position 
+     *                  The position of the player whose hearts will be updated
+     */
+    public HeartsUpdateFunctions(MainActivity act, PlatformPosition position) {
+        this.act = act;
+        this.position = position;
+    }
+    
+   /**
+    * Calls functions to make the hearts the right colour.
+    * @param health the amount of health that has to be displayed in grey
+    *      and red hearts.
+    */
+    public void setHealth(int health) {
+        if (health > 0) {
+            makeRed(1);
+            if (health > 1) {
+                makeRed(2);
+                if (health > 2) {
+                    makeRed(3);
+                }
+                else {
+                    makeGrey(3);
+                }
+            }
+            else {
+                makeGrey(2);
+                makeGrey(3);
+            }
+        }
+        else {
+            makeGrey(1);
+            makeGrey(2);
+            makeGrey(3);
+        }
+    }
+
+    /**
+     * Makes a heart red.
+     * @param heartid the id of the heart which has its sprite changed
+     */
+    public void makeRed(int heartid) {
+        
+        ImageView img = (ImageView) act.findViewById(getHeart(heartid));
+        img.setImageResource(R.drawable.heart_red);
+        
+    }
+
+    /**
+     * Makes a heart grey.
+     * @param heartid the id of the heart which has its sprite changed
+     */
+    public void makeGrey(int heartid) {
+        
+        ImageView img = (ImageView) act.findViewById(getHeart(heartid));
+        img.setImageResource(R.drawable.heart_grey);
+
+    }
+    
+    /**
+     * Returns the right heart to update based on heartid and position.
```

Comment:
I recommend explaining in alittle bit more detail what this method does, because as we just discussed IRL, it's quite complicated with these two different types of IDs.